packages/file_selector/file_selector_web/lib/file_selector_web.dart

```diff
@@ -0,0 +1,222 @@
+import 'dart:async';
+import 'dart:html';
+import 'dart:typed_data';
+
+import 'package:file_selector_platform_interface/file_selector_platform_interface.dart';
+import 'package:flutter_web_plugins/flutter_web_plugins.dart';
+import 'package:meta/meta.dart';
+
+final String _kFileSelectorInputsDomId = '__file_selector_web_file_input';
+
+/// The web implementation of [FileSelectorPlatform].
+///
+/// This class implements the `package:file_selector` functionality for the web.
+class FileSelectorPlugin extends FileSelectorPlatform {
+  
+  /// Open file dialog for loading files and return a XFile
+  @override
+  Future<XFile> loadFile({
+    List<XTypeGroup> acceptedTypeGroups,
+    String initialDirectory,
+    String confirmButtonText,
+  }) {
+    Completer<XFile> _completer = Completer();
+    _loadFileHelper(false, acceptedTypeGroups).then((list) {
+      _completer.complete(list.first);
+    })
+        .catchError((err) {
+      _completer.completeError(err);
+    });
+
+    return _completer.future;
+  }
+
+  /// Open file dialog for loading files and return a XFile
+  @override
+  Future<List<XFile>> loadFiles({
+    List<XTypeGroup> acceptedTypeGroups,
+    String initialDirectory,
+    String confirmButtonText,
+  }) {
+    return _loadFileHelper(true, acceptedTypeGroups);
+  }
+
+  @override
+  Future<String> getSavePath({
+    List<XTypeGroup> acceptedTypeGroups,
+    String initialDirectory,
+    String suggestedName,
+    String confirmButtonText,
+  }) => Future.value();
```

Comment:
If you mark this function as `async` you can just `=> null`, and the language will understand that it's a "Future<String>".